src/stringClass.js

```diff
@@ -1,33 +1,165 @@
 const stringClass = {
+  /**
+   * This method takes a string and tests if it contains vowels
+   * @param {String} this
+   * @return {Boolean} true or false
+   */
   hasVowels() {
     return /[aeiou]/.test(this);
   },
 
-  toUpper() { },
+/**
+   * This method takes a string and returns it in uppercase
+   * @param {String} this
+   * @return {String} this
+   */
+  toUpper() {
+    return this.replace(/[a-z]/g, caps =>
+      String.fromCharCode(caps.charCodeAt(0) - 32)
+    );
+  },
 
-  toLower() { },
+/**
+   * This method takes a string and returns it in lowercase
+   * @param {String} this
+   * @return {String} this
+   */
+  toLower() {
+    return this.replace(/[A-Z]/g, lower =>
+      String.fromCharCode(lower.charCodeAt(0) + 32)
+    );
+  },
 
-  ucFirst() { },
+/**
+   * This method takes a string and returns its first letter in uppercase
+   * @param {String} this
+   * @return {String} this
+   */
+  ucFirst() {
+    return this.replace(this[0], this[0].toUpper());
```

Comment:
This will throw an error for  `ucFirst()` is called on an empty string because then `this[0]` will be `undefined` and calling a method on `undefined` is an error.